draft-ietf-jsonpath-base.md

```diff
@@ -1165,16 +1165,16 @@ The following table lists filter expression operators in order of precedence fro
 |  1  | Logical OR | `¦¦`   |
 {: title="Filter expression operator precedence" }
 
-#### Semantics
+##### Semantics
 {: unnumbered}
 
-The `filter-selector` works with arrays and objects exclusively. Its result is a list of *zero*, *one*, *multiple* or *all* of their array elements or member values, respectively.
+The filter //PICKER// works with arrays and objects exclusively. Its result is a list of *zero*, *one*, *multiple* or *all* of their array elements or member values, respectively.
```

Comment:
The result of a //PICKER// is not a list. This confuses //PICKERS// and selectors.

Reply:
As you can see from the diff, I didn't change this text.  I'm not sure how the result being a nodelist confuses *pickers and selectors given that the result of the child selector is exactly the concatenation of the *pickers it contains.

Reply:
> As you can see from the diff, I didn't change this text.

No, but perhaps you should have. See https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/258#discussion_r967854964 for why.

> 
> I'm not sure how the result being a nodelist confuses *pickers and selectors given that the result of the child selector is exactly the concatenation of the *pickers it contains.

If the result of a *picker was simply a boolean (match or didn't match), there would be a clearer distinction between selectors and *pickers.

Reply:
The numeric index *picker `1` doesn't have a boolean result.  It has a value result.  The slice *picker has a set of values as a result.  Similarly, the filter *picker has a set of values as a result.

Saying a *picker has a boolean result may work for filters, but it doesn't work for the other *pickers.  This confuses what a *picker is.

Reply:
Not necessarily. Option 2 allows other *pickers to function correctly because they can use the name or index of the value as part of their spec.